gestures.md

```diff
@@ -0,0 +1,102 @@
+widget to the root of the tree. There is no mechanism for canceling or stopping
+pointer events from being dispatched further.
+
+# Gestures
+
+Gestures represent semantic actions (e.g., tap, drag, and scale) that are
+recognized from multiple individual pointer events, potentially even multiple
+individual pointers. Gestures can dispatch multiple events, corresponding to the
+lifecycle of the gesture (e.g., drag start, drag update, and drag end):
+
+- Tap
+ - `onTapDown` A pointer that might cause a tap has contacted the screen at a
+   particular location.
+ - `onTapUp` A pointer that will trigger a tap has stopped contacting the screen
+   at a particular location.
+ - `onTap` A tap has occurred.
+ - `onTapCancel` The pointer that previously triggered the `onTapDown` will not
+   end up causing a tap.
+- Double tap
+ - `onDoubleTap` The user has tapped the screen at the same location twice in
+   quick succession.
+- Long press
+ - `onLongPress` A pointer has remained in contact with the screen at the same
+   location for a long period of time.
+- Vertical drag
+ - `onVerticalDragStart` A pointer has contacted the screen and might begin to
+   move vertically.
+ - `onVerticalDragUpdate` A pointer that is in contact with the screen and
+   moving vertically has moved again in the vertical direction.
+ - `onVerticalDragEnd` A pointer that was previously in contact with the screen
+   and moving vertically is no longer in contact with the screen and was moving
+   at a specific velocity when it stopped contacting the screen.
+- Horizontal drag
+ - `onHorizontalDragStart` A pointer has contacted the screen and might begin to
+   move horizontally.
+ - `onHorizontalDragUpdate` A pointer that is in contact with the screen and
+   moving horizontally has moved again in the horizontal direction.
+ - `onHorizontalDragEnd` A pointer that was previously in contact with the
+   screen and moving horizontally is no longer in contact with the screen and
+   was moving at a specific velocity when it stopped contacting the screen.
+
+## Gesture disambiguation
+
+At a given location on screen, there might be multiple gesture detectors. All
+of these gesture detectors listen to the stream of pointer events as they flow
+past and attempt to recognize specific gestures. The `GestureDetector` widget
+decides which gestures to attempt to recognize based on which of its callbacks
+are non-null.
+
+When there is more than one gesture recognizer for a given pointer on the
+screen, the framework disambiguates which gesture the user intends by having
+each recognizer join the _gesture arena_. The gesture arena determines which
+gesture wins using the following rules:
+
+- At any time, a recognizer can declare defeat and leave the arena.  If there's
+  only one recognizer left in the arena, that recognizer is the winner.
+
+- At any time, a recognizer can declare victory, which causes it to win and all
+  the remaining recognizers to lose.
+
```

Comment:
We should probably talk about holding the arena open, and how it gets swept when the pointer goes up, and how a single gesture can be in multiple arenas since it can track multiple pointers...